File: android/app/src/main/java/com/rork/turnoevandroid/ui/screens/RecoveryProgramSection.kt
package com.rork.turnoevandroid.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CalendarMonth
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.ChevronLeft
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.filled.EventAvailable
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.rork.turnoevandroid.FleetViewModel
import com.rork.turnoevandroid.data.FleetState
import com.rork.turnoevandroid.domain.BonusKind
import com.rork.turnoevandroid.domain.BonusRules
import com.rork.turnoevandroid.domain.ShiftRules
import com.rork.turnoevandroid.domain.ShiftSlot
import com.rork.turnoevandroid.ui.components.BigButton
import com.rork.turnoevandroid.ui.theme.CapsLabel
import com.rork.turnoevandroid.ui.theme.Palette
import com.rork.turnoevandroid.ui.theme.TabularNumbers
import com.rork.turnoevandroid.ui.theme.panel
import com.rork.turnoevandroid.ui.theme.panelFlat
import com.rork.turnoevandroid.util.Fmt
import com.rork.turnoevandroid.util.plusDays
import com.rork.turnoevandroid.util.plusMonths
import java.util.Calendar

/**
 * Bonus recovery program: the driver reserves days on the opposite group
 * (weekday drivers on Saturday/Sunday, weekend drivers Monday to Friday)
 * and picks the morning or evening slot.
 */
@Composable
fun RecoveryProgramSection(
    viewModel: FleetViewModel,
    state: FleetState,
    now: Long,
    suggestedBonus: BonusKind,
    modifier: Modifier = Modifier,
) {
    val driver = viewModel.driver
    var monthAnchor by remember { mutableStateOf<Long?>(null) }
    var selectedDay by remember { mutableStateOf<Long?>(null) }
    var slot by remember { mutableStateOf<ShiftSlot?>(null) }
    var bonus by remember { mutableStateOf<BonusKind?>(null) }
    var feedback by remember { mutableStateOf<String?>(null) }

    val anchor = monthAnchor ?: BonusRules.monthStart(now)
    val activeSlot = slot ?: driver.slot
    val activeBonus = bonus ?: suggestedBonus

    Column(
        modifier
            .fillMaxWidth()
            .panel()
            .padding(18.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp),
    ) {
        // Header
        Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            Row(horizontalArrangement = Arrangement.spacedBy(10.dp), verticalAlignment = Alignment.CenterVertically) {
                Box(
                    Modifier
                        .size(42.dp)
                        .clip(RoundedCornerShape(14.dp))
                        .background(Palette.volt.copy(alpha = 0.12f)),
                    contentAlignment = Alignment.Center,
                ) {
                    Icon(Icons.Filled.EventAvailable, null, Modifier.size(20.dp), tint = Palette.volt)
                }
                Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
                    Text(
                        "Programa de recuperación de bonos",
                        style = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Black, color = Palette.textPrimary),
                    )
                    Text(
                        "Turnos disponibles: ${BonusRules.recoveryDaysLabel(driver)}",
                        style = TextStyle(fontSize = 11.sp, color = Palette.textMuted),
                    )
                }
            }
            Text(
                "Tu grupo es ${driver.group.label.lowercase()}, así que recuperas en " +
                    "${BonusRules.recoveryGroup(driver).label.lowercase()}. Elige el día y el turno en el " +
                    "que quieres laborar.",
                style = TextStyle(fontSize = 11.sp, color = Palette.textMuted),
            )
        }

        // Slot picker
        Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            CapsLabel("Turno a laborar")
            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                ShiftSlot.entries.forEach { option ->
                    val isActive = activeSlot == option
                    Column(
                        Modifier
                            .weight(1f)
                            .clip(RoundedCornerShape(16.dp))
                            .background(
                                if (isActive) {
                                    Palette.volt.copy(alpha = 0.13f)
                                } else {
                                    Palette.surfaceRaised.copy(alpha = 0.6f)
                                },
                            )
                            .border(
                                1.dp,
                                if (isActive) Palette.volt.copy(alpha = 0.6f) else Palette.hairline,
                                RoundedCornerShape(16.dp),
                            )
                            .clickable { slot = option }
                            .padding(12.dp),
                        verticalArrangement = Arrangement.spacedBy(2.dp),
                    ) {
                        Text(
                            option.label,
                            style = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Palette.textPrimary),
                        )
                        Text(
                            option.rangeLabel,
                            style = TextStyle(
                                fontSize = 10.sp,
                                fontFamily = TabularNumbers,
                                color = Palette.textMuted,
                            ),
                        )
                    }
                }
            }
        }

        // Bonus picker
        Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            CapsLabel("Bono a recuperar")
            Row(
                Modifier
                    .fillMaxWidth()
                    .horizontalScroll(rememberScrollState()),
                horizontalArrangement = Arrangement.spacedBy(8.dp),
            ) {
                BonusKind.entries.filterNot { it.isExternal }.forEach { option ->
                    val isActive = activeBonus == option
                    Text(
                        option.title,
                        style = TextStyle(
                            fontSize = 11.sp,
                            fontWeight = FontWeight.Bold,
                            color = if (isActive) Palette.canvas else Palette.textMuted,
                        ),
                        modifier = Modifier
                            .clip(CircleShape)
                            .background(if (isActive) Palette.volt else Palette.surfaceRaised.copy(alpha = 0.7f))
                            .border(
                                1.dp,
                                if (isActive) Color.Transparent else Palette.hairline,
                                CircleShape,
                            )
                            .clickable { bonus = option }
                            .padding(horizontal = 12.dp, vertical = 9.dp),
                    )
                }
            }
        }

        // Calendar
        Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
            Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
                CalendarArrow(Icons.Filled.ChevronLeft) {
                    monthAnchor = anchor.plusMonths(-1)
                    selectedDay = null
                }
                Spacer(Modifier.weight(1f))
                Text(
                    Fmt.monthLong(anchor),
                    style = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Black, color = Palette.textPrimary),
                )
                Spacer(Modifier.weight(1f))
                CalendarArrow(Icons.Filled.ChevronRight) {
                    monthAnchor = anchor.plusMonths(1)
                    selectedDay = null
                }
            }

            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                listOf("L", "M", "M", "J", "V", "S", "D").forEach { day ->
                    Text(
                        day,
                        style = TextStyle(
                            fontSize = 10.sp,
                            fontWeight = FontWeight.Black,
                            color = Palette.textMuted,
                        ),
                        textAlign = TextAlign.Center,
                        modifier = Modifier.weight(1f),
                    )
                }
            }

            monthCells(anchor).chunked(7).forEach { week ->
                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                    week.forEach { day ->
                        if (day == null) {
                            Spacer(
                                Modifier
                                    .weight(1f)
                                    .height(42.dp),
                            )
                        } else {
                            val booking = viewModel.recoveryBooking(day)
                            val isAvailable = BonusRules.canBook(driver, day, now)
                            val isSelected = selectedDay?.let { ShiftRules.isSameDay(it, day) } == true
                            val isToday = ShiftRules.isSameDay(day, now)
                            DayCell(
                                dayNumber = Fmt.calendar(day).get(Calendar.DAY_OF_MONTH),
                                isSelected = isSelected,
                                isAvailable = isAvailable,
                                hasBooking = booking != null,
                                isToday = isToday,
                                modifier = Modifier.weight(1f),
                            ) {
                                if (isAvailable && booking == null) {
                                    selectedDay = day
                                    feedback = null
                                }
                            }
                        }
                    }
                    repeat(7 - week.size) {
                        Spacer(
                            Modifier
                                .weight(1f)
                                .height(42.dp),
                        )
                    }
                }
            }
        }

        // Footer
        Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
            val chosen = selectedDay
            if (chosen != null) {
                Row(
                    Modifier
                        .fillMaxWidth()
                        .panelFlat()
                        .padding(12.dp),
                    horizontalArrangement = Arrangement.spacedBy(8.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Icon(Icons.Filled.CalendarMonth, null, Modifier.size(14.dp), tint = Palette.volt)
                    Text(
                        "${Fmt.dateShort(chosen)} · ${activeSlot.label} ${activeSlot.rangeLabel}",
                        style = TextStyle(fontSize = 11.sp, fontWeight = FontWeight.SemiBold, color = Palette.textPrimary),
                    )
                }
            } else {
                Text(
                    "Selecciona un día disponible (marcado con punto) para reservar tu lugar.",
                    style = TextStyle(fontSize = 11.sp, color = Palette.textMuted),
                )
            }

            feedback?.let { message ->
                Text(message, style = TextStyle(fontSize = 13.sp, color = Palette.volt))
            }

            BigButton(
                title = "Reservar día de recuperación",
                icon = Icons.Filled.CheckCircle,
                enabled = chosen != null,
            ) {
                val day = chosen ?: return@BigButton
                val saved = viewModel.bookRecovery(day, activeSlot, activeBonus)
                feedback = if (saved) {
                    selectedDay = null
                    "Reserva confirmada para ${Fmt.dateShort(day)} en turno ${activeSlot.label.lowercase()}."
                } else {
                    "Ese día ya no está disponible. Elige otro."
                }
            }
        }

        if (state.recoveryBookings.isNotEmpty()) {
            Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                CapsLabel("Mis reservas")
                viewModel.upcomingRecoveryBookings.forEach { booking ->
                    Row(
                        Modifier
                            .fillMaxWidth()
                            .panelFlat()
                            .padding(12.dp),
                        horizontalArrangement = Arrangement.spacedBy(10.dp),
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
                            Text(
                                Fmt.dateShort(booking.date),
                                style = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Bold, color = Palette.textPrimary),
                            )
                            Text(
                                "${booking.slot.label} ${booking.slot.rangeLabel} · bono de ${booking.bonus.shortName}",
                                style = TextStyle(fontSize = 10.sp, color = Palette.textMuted),
                            )
                        }
                        Text(
                            "Cancelar",
                            style = TextStyle(fontSize = 10.sp, fontWeight = FontWeight.Bold, color = Palette.danger),
                            modifier = Modifier
                                .clip(CircleShape)
                                .background(Palette.danger.copy(alpha = 0.12f))
                                .clickable { viewModel.cancelRecovery(booking.id) }
                                .padding(horizontal = 10.dp, vertical = 6.dp),
                        )
                    }
                }
            }
        }
    }
}

@Composable
private fun CalendarArrow(icon: androidx.compose.ui.graphics.vector.ImageVector, onClick: () -> Unit) {
    Box(
        Modifier
            .size(34.dp)
            .background(Palette.surfaceRaised, CircleShape)
            .clickable { onClick() },
        contentAlignment = Alignment.Center,
    ) {
        Icon(icon, null, Modifier.size(18.dp), tint = Palette.textPrimary)
    }
}

@Composable
private fun DayCell(
    dayNumber: Int,
    isSelected: Boolean,
    isAvailable: Boolean,
    hasBooking: Boolean,
    isToday: Boolean,
    modifier: Modifier = Modifier,
    onClick: () -> Unit,
) {
    val foreground = when {
        isSelected -> Palette.canvas
        hasBooking -> Palette.volt
        isAvailable -> Palette.textPrimary
        else -> Palette.textMuted.copy(alpha = 0.45f)
    }
    val background = when {
        isSelected -> Palette.volt
        hasBooking -> Palette.volt.copy(alpha = 0.16f)
        isAvailable -> Palette.surfaceRaised.copy(alpha = 0.85f)
        else -> Palette.surfaceRaised.copy(alpha = 0.25f)
    }
    Column(
        modifier
            .height(42.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(background)
            .then(
                if (isToday) {
                    Modifier.border(1.5.dp, Palette.info.copy(alpha = 0.7f), RoundedCornerShape(12.dp))
                } else {
                    Modifier
                },
            )
            .clickable(enabled = isAvailable && !hasBooking) { onClick() },
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        Text(
            "$dayNumber",
            style = TextStyle(
                fontSize = 13.sp,
                fontWeight = FontWeight.Bold,
                fontFamily = TabularNumbers,
                color = foreground,
            ),
        )
        when {
            hasBooking -> Icon(Icons.Filled.Check, null, Modifier.size(10.dp), tint = foreground)
            isAvailable -> Box(
                Modifier
                    .size(4.dp)
                    .background(if (isSelected) Palette.canvas else Palette.volt, CircleShape),
            )
        }
    }
}

/** Leading blanks + every day of the anchored month, Monday first. */
private fun monthCells(anchor: Long): List<Long?> {
    val calendar = Fmt.calendar(anchor)
    val daysInMonth = calendar.getActualMaximum(Calendar.DAY_OF_MONTH)
    val weekday = calendar.get(Calendar.DAY_OF_WEEK)
    val leading = (weekday + 5) % 7
    val cells = mutableListOf<Long?>()
    repeat(leading) { cells += null }
    for (offset in 0 until daysInMonth) {
        cells += anchor.plusDays(offset)
    }
    return cells
}

File: ios-turno-ev/TurnoEV/Views/RecoveryProgramSection.swift
import SwiftUI

/// Bonus recovery program: the driver reserves days on the opposite group
/// (weekday drivers on Saturday/Sunday, weekend drivers Monday to Friday)
/// and picks the morning or evening slot.
struct RecoveryProgramSection: View {
    @Environment(FleetStore.self) private var store

    let suggestedBonus: BonusKind

    /// Start of the logical day, resolved once by the host screen.
    ///
    /// Everything this calendar asks of the clock is day-granular — which month it opens
    /// on, which cells are still bookable, which one is today — so it takes the day instead
    /// of reading `store.now` in its own body. The host owns the one detector; this section
    /// simply redraws when the day it was given changes.
    let today: Date

    @State private var monthAnchor: Date?
    @State private var selectedDay: Date?
    @State private var slot: ShiftSlot?
    @State private var bonus: BonusKind?
    @State private var feedback: String?

    private var anchor: Date { monthAnchor ?? BonusRules.monthStart(for: today) }
    private var activeSlot: ShiftSlot { slot ?? store.driver.slot }
    private var activeBonus: BonusKind { bonus ?? suggestedBonus }

    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            header
            slotPicker
            bonusPicker
            calendar
            selectionFooter

            if !store.recoveryBookings.isEmpty {
                bookingsList
            }
        }
        .padding(18)
        .panel()
    }

    // MARK: - Header

    private var header: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack(spacing: 10) {
                Image(systemName: "calendar.badge.plus")
                    .font(.system(.body, weight: .semibold))
                    .foregroundStyle(Palette.volt)
                    .frame(width: 42, height: 42)
                    .background(Palette.volt.opacity(0.12), in: .rect(cornerRadius: 14))
                VStack(alignment: .leading, spacing: 2) {
                    Text("Programa de recuperación de bonos")
                        .font(.system(.subheadline, weight: .black))
                    Text("Turnos disponibles: \(BonusRules.recoveryDaysLabel(for: store.driver))")
                        .font(.caption2)
                        .foregroundStyle(Palette.textMuted)
                }
                Spacer(minLength: 0)
            }

            Text("Tu grupo es \(store.driver.group.label.lowercased()), así que recuperas en \(BonusRules.recoveryGroup(for: store.driver).label.lowercased()). Elige el día y el turno en el que quieres laborar.")
                .font(.system(size: 11))
                .foregroundStyle(Palette.textMuted)
        }
    }

    private var slotPicker: some View {
        VStack(alignment: .leading, spacing: 8) {
            CapsLabel(text: "Turno a laborar")
            HStack(spacing: 10) {
                ForEach(ShiftSlot.allCases, id: \.self) { option in
                    let isActive = activeSlot == option
                    Button {
                        slot = option
                    } label: {
                        VStack(alignment: .leading, spacing: 2) {
                            Text(option.label)
                                .font(.system(.subheadline, weight: .bold))
                            Text(option.rangeLabel)
                                .font(.system(size: 10))
                                .monospacedDigit()
                                .foregroundStyle(Palette.textMuted)
                        }
                        .frame(maxWidth: .infinity, alignment: .leading)
                        .padding(12)
                        .background((isActive ? Palette.volt.opacity(0.13) : Palette.surfaceRaised.opacity(0.6)), in: .rect(cornerRadius: 16))
                        .overlay {
                            RoundedRectangle(cornerRadius: 16)
                                .stroke(isActive ? Palette.volt.opacity(0.6) : Palette.hairline, lineWidth: 1)
                        }
                    }
                    .buttonStyle(.plain)
                }
            }
        }
    }

    private var bonusPicker: some View {
        VStack(alignment: .leading, spacing: 8) {
            CapsLabel(text: "Bono a recuperar")
            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: 8) {
                    ForEach(BonusKind.allCases.filter { !$0.isExternal }) { option in
                        let isActive = activeBonus == option
                        Button {
                            bonus = option
                        } label: {
                            HStack(spacing: 5) {
                                Image(systemName: option.symbol)
                                    .font(.system(size: 10, weight: .bold))
                                Text(option.title)
                                    .font(.system(size: 11, weight: .bold))
                            }
                            .foregroundStyle(isActive ? Palette.canvas : Palette.textMuted)
                            .padding(.horizontal, 12)
                            .padding(.vertical, 9)
                            .background(isActive ? Palette.volt : Palette.surfaceRaised.opacity(0.7), in: .capsule)
                            .overlay { Capsule().stroke(isActive ? .clear : Palette.hairline, lineWidth: 1) }
                        }
                        .buttonStyle(.plain)
                    }
                }
            }
            .contentMargins(.horizontal, 0)
        }
    }

    // MARK: - Calendar

    private var calendar: some View {
        VStack(spacing: 12) {
            HStack {
                Button {
                    shiftMonth(by: -1)
                } label: {
                    Image(systemName: "chevron.left")
                        .font(.system(.footnote, weight: .bold))
                        .frame(width: 34, height: 34)
                        .background(Palette.surfaceRaised, in: .circle)
                }
                .buttonStyle(.plain)

                Spacer()
                Text(Fmt.monthLong(anchor))
                    .font(.system(.subheadline, weight: .black))
                Spacer()

                Button {
                    shiftMonth(by: 1)
                } label: {
                    Image(systemName: "chevron.right")
                        .font(.system(.footnote, weight: .bold))
                        .frame(width: 34, height: 34)
                        .background(Palette.surfaceRaised, in: .circle)
                }
                .buttonStyle(.plain)
            }

            HStack(spacing: 4) {
                ForEach(["L", "M", "M", "J", "V", "S", "D"], id: \.self) { day in
                    Text(day)
                        .font(.system(size: 10, weight: .black))
                        .foregroundStyle(Palette.textMuted)
                        .frame(maxWidth: .infinity)
                }
            }

            LazyVGrid(columns: Array(repeating: GridItem(.flexible(), spacing: 4), count: 7), spacing: 4) {
                ForEach(Array(monthCells.enumerated()), id: \.offset) { _, day in
                    if let day {
                        dayCell(day)
                    } else {
                        Color.clear.frame(height: 42)
                    }
                }
            }
        }
    }

    private var monthCells: [Date?] {
        let cal = ShiftRules.calendar
        guard let range = cal.range(of: .day, in: .month, for: anchor) else { return [] }
        let weekday = cal.component(.weekday, from: anchor)
        let leading = (weekday + 5) % 7
        var cells: [Date?] = Array(repeating: nil, count: leading)
        for offset in 0..<range.count {
            cells.append(cal.date(byAdding: .day, value: offset, to: anchor))
        }
        return cells
    }

    private func dayCell(_ day: Date) -> some View {
        let cal = ShiftRules.calendar
        let booking = store.recoveryBooking(on: day)
        let isAvailable = BonusRules.canBook(driver: store.driver, date: day, now: today)
        let isSelected = selectedDay.map { cal.isDate($0, inSameDayAs: day) } ?? false
        let isToday = ShiftRules.isSameDay(day, today)
        let dayNumber = cal.component(.day, from: day)

        return Button {
            guard isAvailable, booking == nil else { return }
            selectedDay = day
            feedback = nil
        } label: {
            VStack(spacing: 2) {
                Text("\(dayNumber)")
                    .font(.system(size: 13, weight: .bold))
                    .monospacedDigit()
                if booking != nil {
                    Image(systemName: "checkmark")
                        .font(.system(size: 8, weight: .black))
                } else if isAvailable {
                    Circle().fill(Palette.volt).frame(width: 4, height: 4)
                }
            }
            .foregroundStyle(foreground(isSelected: isSelected, isAvailable: isAvailable, hasBooking: booking != nil))
            .frame(maxWidth: .infinity)
            .frame(height: 42)
            .background(background(isSelected: isSelected, isAvailable: isAvailable, hasBooking: booking != nil), in: .rect(cornerRadius: 12))
            .overlay {
                RoundedRectangle(cornerRadius: 12)
                    .stroke(isToday ? Palette.info.opacity(0.7) : .clear, lineWidth: 1.5)
            }
        }
        .buttonStyle(.plain)
        .disabled(!isAvailable || booking != nil)
    }

    private func foreground(isSelected: Bool, isAvailable: Bool, hasBooking: Bool) -> Color {
        if isSelected { return Palette.canvas }
        if hasBooking { return Palette.volt }
        return isAvailable ? .primary : Palette.textMuted.opacity(0.45)
    }

    private func background(isSelected: Bool, isAvailable: Bool, hasBooking: Bool) -> Color {
        if isSelected { return Palette.volt }
        if hasBooking { return Palette.volt.opacity(0.16) }
        return isAvailable ? Palette.surfaceRaised.opacity(0.85) : Palette.surfaceRaised.opacity(0.25)
    }

    private func shiftMonth(by value: Int) {
        let cal = ShiftRules.calendar
        guard let next = cal.date(byAdding: .month, value: value, to: anchor) else { return }
        monthAnchor = next
        selectedDay = nil
    }

    // MARK: - Footer

    private var selectionFooter: some View {
        VStack(alignment: .leading, spacing: 10) {
            if let selectedDay {
                HStack(spacing: 8) {
                    Image(systemName: "calendar")
                        .font(.caption2)
                        .foregroundStyle(Palette.volt)
                    Text("\(Fmt.dateShort(selectedDay).capitalized) · \(activeSlot.label) \(activeSlot.rangeLabel)")
                        .font(.system(size: 11, weight: .semibold))
                    Spacer(minLength: 0)
                }
                .padding(12)
                .frame(maxWidth: .infinity, alignment: .leading)
                .panelFlat()
            } else {
                Text("Selecciona un día disponible (marcado con punto) para reservar tu lugar.")
                    .font(.system(size: 11))
                    .foregroundStyle(Palette.textMuted)
            }

            if let feedback {
                Text(feedback)
                    .font(.footnote)
                    .foregroundStyle(Palette.volt)
            }

            BigButton(
                title: "Reservar día de recuperación",
                symbol: "checkmark.circle.fill",
                isEnabled: selectedDay != nil
            ) {
                reserve()
            }
        }
    }

    private func reserve() {
        guard let day = selectedDay else { return }
        let saved = store.bookRecovery(date: day, slot: activeSlot, bonus: activeBonus)
        if saved {
            feedback = "Reserva confirmada para \(Fmt.dateShort(day).capitalized) en turno \(activeSlot.label.lowercased())."
            selectedDay = nil
        } else {
            feedback = "Ese día ya no está disponible. Elige otro."
        }
    }

    private var bookingsList: some View {
        VStack(alignment: .leading, spacing: 10) {
            CapsLabel(text: "Mis reservas")
            ForEach(store.upcomingRecoveryBookings) { booking in
                HStack(spacing: 10) {
                    VStack(alignment: .leading, spacing: 2) {
                        Text(Fmt.dateShort(booking.date).capitalized)
                            .font(.system(.caption, weight: .bold))
                        Text("\(booking.slot.label) \(booking.slot.rangeLabel) · bono de \(booking.bonus.shortName)")
                            .font(.system(size: 10))
                            .foregroundStyle(Palette.textMuted)
                    }
                    Spacer(minLength: 0)
                    Button {
                        store.cancelRecovery(id: booking.id)
                    } label: {
                        Text("Cancelar")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundStyle(Palette.danger)
                            .padding(.horizontal, 10)
                            .padding(.vertical, 6)
                            .background(Palette.danger.opacity(0.12), in: .capsule)
                    }
                    .buttonStyle(.plain)
                }
                .padding(12)
                .panelFlat()
            }
        }
    }
}

#Preview {
    ScrollView {
        RecoveryProgramSection(suggestedBonus: .punctuality, today: .now)
            .padding()
    }
    .background { StationBackground() }
    .environment(FleetStore())
    .preferredColorScheme(.dark)
}
